src/platform/ESP32/MdnsBasicPacketFilters.h

```diff
@@ -0,0 +1,112 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <atomic>
+#include <inet/EndpointQueueFilter.h>
+#include <inet/IPPacketInfo.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace mdns {
+namespace Minimal {
+
+/**
+ * @brief Basic filter that counts how many pending (not yet dequeued) mDNS packets
+ *        are accumulated, and drops those that would cause crossing of the threshold.
+ *
+ */
+class DropIfTooManyQueuedMdnsPacketFilter : public chip::Inet::EndpointQueueFilter
+{
+    // Repeated here since there are no accessible headers with this
+    static constexpr uint16_t kMdnsPort = 5353u;
+
+public:
+    /**
+     * @brief Initialize the packet filter with a starting limit
+     *
+     * @param maxQueuedPackets - max number of pending-in-queue mDNS packets not yet processed
+     */
+    DropIfTooManyQueuedMdnsPacketFilter(size_t maxQueuedPackets) : mMaxQueuedPackets(maxQueuedPackets) {}
+
+    FilterOutcome FilterBeforeEnqueue(const void * endpoint, const chip::Inet::IPPacketInfo & pktInfo,
+                                      const chip::System::PacketBufferHandle & pktPayload) override
+    {
+        // WARNING: This is likely called in a different context than `FilterAfterDequeue`. We use an atomic for the counter.
+
+        // Non-mDNS is not filtered
+        VerifyOrReturnValue(pktInfo.DestPort == kMdnsPort, FilterOutcome::kAllowPacket);
+
+        if (pktInfo.DestAddress.IsIPv4() && pktInfo.DestPort == kMdnsPort)
+        {
+            if (!mStopLogging)
+            {
+                ChipLogError(Discovery, "Started dropping mDNS from storm.");
```

Comment:
This case has nothing to do with "storm",  This is going to log as soon as we get an ipv4 packet, then stop logging usefully....